core/conversion/converters/impl/plugins/interpolate_plugin.cpp

```diff
@@ -0,0 +1,244 @@
+
+    output_archive.write("in_shape", torch::IValue(in_shape));
+    output_archive.write("out_shape", torch::IValue(out_shape));
+    output_archive.write("size", torch::IValue(size));
+    output_archive.write("mode", torch::IValue(mode));
+    output_archive.write("align_corners", torch::IValue(align_corners));
+
+    std::ostringstream data_str;
+    output_archive.save_to(data_str);
+
+    return data_str.str();
+}
+
+size_t InterpolatePlugin::getSerializationSize() const {
+    return serializeToString().size();
+}
+
+bool InterpolatePlugin::supportsFormatCombination(int pos, const nvinfer1::PluginTensorDesc* inOut, int nbInputs, int nbOutputs) {
+    if (inOut->format != nvinfer1::TensorFormat::kLINEAR) {
+        return false;
+    } 
+
+    if (inOut->type == DataType::kINT32 || inOut->type == DataType::kINT8) {
+        return false;
+    }
+
+    return true;
+}
+
+void InterpolatePlugin::configurePlugin(const nvinfer1::DynamicPluginTensorDesc* in, int nbInputs, const nvinfer1::DynamicPluginTensorDesc* out, int nbOutputs) {
+    dtype = DataType::kFLOAT;
+}
+
+size_t InterpolatePlugin::getWorkspaceSize(const nvinfer1::PluginTensorDesc* inputs, int nbInputs, const nvinfer1::PluginTensorDesc* outputs, int nbOutputs) const {
+    return 0;
+}
+
+int InterpolatePlugin::enqueue(const nvinfer1::PluginTensorDesc* inputDesc, const nvinfer1::PluginTensorDesc* outputDesc, const void *const *inputs, 
+                                                                                                        void *const *outputs, void *workspace, 
+                                                                                                        cudaStream_t stream) {
+    at::Tensor input;
+
+    if (mode == "adaptive_pool2d") {
+        // use dynamically inferred input shape (for pooling)
+        input = at::from_blob((void*) inputs[0], util::toVec(inputDesc->dims), [](void*){}, tensor_options);
+    } else {
+        // use precomputed input shape (for interpolation/upsampling)
+        input = at::from_blob((void*) inputs[0], in_shape, [](void*){}, tensor_options);
+    }
+
+    at::Tensor output = at::from_blob(outputs[0], out_shape, [](void*){}, tensor_options);
+
+    at::cuda::CUDAStream torch_stream = at::cuda::getStreamFromPool();
+    at::cuda::CUDAStreamGuard torch_guard(torch_stream);
+
+    cudaEvent_t event;
+    cudaEventCreate(&event);
+    cudaEventRecord(event, stream);
+
+    cudaStreamWaitEvent(torch_stream.stream(), event, 0);
+
+    if (mode == "linear") {
+        at::upsample_linear1d_out(output, input, {size[0]}, align_corners);
+    } else if (mode == "bilinear") {
+        at::upsample_bilinear2d_out(output, input, {size[0], size[1]}, align_corners);
+    } else if (mode == "trilinear") {
+        at::upsample_trilinear3d_out(output, input, {size[0], size[1], size[2]}, align_corners);
+    } else if (mode == "adaptive_pool2d") {
+        at::adaptive_avg_pool2d_out(output, input, {size[0], size[1]});

```

Comment:
There is a know issues section of the README that talks about adaptive_avg_pool2d dynamic, can you delete it now that it is fixed?

Reply:
Added code snippet you gave, works for FP16 now